server/mailet/mailets/src/test/java/org/apache/james/transport/mailets/SubAddressingTest.java

```diff
@@ -96,40 +97,79 @@ void shouldNotAddStorageDirectiveWhenNoRight() throws Exception {
     @Test
     void shouldAddStorageDirectiveMatchingDetailsWhenRight() throws Exception {
 
-        MailboxSession session = mailboxManager.createSystemSession(Username.of("recipient"));
+        Username username = Username.of("recipient");
+        MailboxSession session = mailboxManager.createSystemSession(username);
 
         // create Mailbox
         MailboxId mailboxId = mailboxManager.createMailbox(
-            MailboxPath.forUser(Username.of("recipient"), "any"), session)
-            .get();
+            MailboxPath.forUser(username, "any"), session).get();
 
-        // give right to post
-        MailboxACL.ACLCommand command = MailboxACL.command()
-            .key(MailboxACL.ANYBODY_KEY)
-            .rights(MailboxACL.Right.Post)
-            .asAddition();
+        giveRightToPost(mailboxId, session);
 
-        mailboxManager.applyRightsCommand(mailboxId, command, session);
+        Mail mail = sendSubAddressedMail(true);
+
+        AttributeName recipient = AttributeName.of("DeliveryPaths_recipient@localhost");
+        assertThat(mail.attributes().map(this::unbox))
+            .doesNotContain(Pair.of(recipient, "any"));
+    }
+
+    @Test
+    void shouldAddStorageDirectiveWhenEveryoneHasRightAndSenderIsUnknown() throws Exception {
+
+        Username username = Username.of("recipient");
+        MailboxSession session = mailboxManager.createSystemSession(username);
```

Comment:
Relocate this boiler plate in @beforeEach, creating related class field as needed